### DISCUSSION TITLE
[Enhancement]: specify hvac_mode to use when window open is detected.

### DISCUSSION THREAD

**Discussion by Abyssmal** [jmcollin78/versatile_thermostat#254] [Ideas]
Firstly - thank you for your great work! It's amazing how much thought went into this project. I will be making a donation in a bit.
I've been using VTherm for 2 months now (start of heating season here). I've incorporated door/window sensors, motion/presence, power meters on the hvacs. Everything! Also using it over two convector heaters. And it's working flawlessly. Energy bill went 15% down, maybe more.
In cold season as can be expected windows and doors are being opened just a few times a day to freshen up the air. I've made some node red automations to switch mode to fan_only when window is open and VTherm turns off, so it ventilates the room faster. And after closing - switching to heat mode.

**Problem** is - the integration turns completely off the hvacs on window open, which can significantly impact the life of the hvacs - closing and opening the covers.

**Proposal**: add option to _**change hvac mode**_ on window open, instead of turning off.

Again keep up the great work, my friend. You, sir, are a time saver!

PS: Sorry for incomplete topic title.

**Comment by jmcollin78** [comment]
Hello @Abyssmal,

First thank you for the compliments. It is very pleasing after this hard works !

> add option to change hvac mode on window open, instead of turning off.

Turning off is changing the hvac_mode to off. Hvac-mode are Heat/Cool/Off/Fan...

Maybe you want to speak about preset (and not hvac_modes). Changing Preset could be an option when window is detected open. That is true. But preset is more to change the target temp and not to stop heating. More over presets are used by internal automations so there is a risk of conflict when using preset. Example: what to do if a movement is no more detected (preset changed to let say Confort) but window is open (and perset should be to Eco) ?
So it is not a good option I think.

**Comment by Abyssmal** [comment]
I do not mean presets. As stated - change HVACMode to fan_only (in my case) instead of off.

I looked through your code and can see self.async_set_hvac_mode(HVACMode.OFF) called on window open.
If that can be changed to take a input variable from config flow without much hussle it'd be great.

**Reply by jmcollin78** [reply]
Yes this could be an option. I was not sure of the need.

**Reply by jmcollin78** [reply]
If you could write a PR it would be super great !

**Reply by jmcollin78** [reply]
I open an issue referencing this discussion.

**Reply by Abyssmal** [reply]
Thank you for understanding.
Sadly, I am not a programmer, just basic knowledge. If you deem this feature is nice to have and got the time to implement it - it would be great.

Again - this is great work on the algorithms!

**Comment by raphpa** [comment]
Not sure if this is a similar situation. I would like to set the thermostat to a (low) target temperature on window open.

The Aqara TRV E1 does a full valve cycle (moving to full open, then to full close) when you set it to Off. This obviously drains a lot of battery. Setting it to a low target temperature would just make it close the valve directly.

**Reply by jmcollin78** [reply]
I seems a similar need indeed. Can you add this comment to this issue so that I will take it into account: https://github.com/jmcollin78/versatile_thermostat/issues/255
